src/node_url.cc

```diff
@@ -1351,11 +1362,47 @@ namespace url {
                             v8::NewStringType::kNormal).ToLocalChecked());
   }
 
-  static void DomainToASCII(const FunctionCallbackInfo<Value>& args) {
+  static void ToUSVString(const FunctionCallbackInfo<Value>& args) {
     Environment* env = Environment::GetCurrent(args);
-    CHECK_GE(args.Length(), 1);
+    CHECK_GE(args.Length(), 2);
     CHECK(args[0]->IsString());
-    Utf8Value value(env->isolate(), args[0]);
+    CHECK(args[1]->IsNumber());
+
+    TwoByteValue value(env->isolate(), args[0]);
+    const size_t n = value.length();
+
+    const int64_t start = args[1]->IntegerValue(env->context()).FromJust();
+    CHECK_GE(start, 0);
+
+    for (size_t i = start; i < n; i++) {
+      uint16_t c = value[i];
+      if (!U16_IS_SURROGATE(c)) {
+        continue;
+      } else if (U16_IS_SURROGATE_TRAIL(c) || i == n - 1) {
+        value[i] = UNICODE_REPLACEMENT_CHARACTER;
+      } else {
+        uint16_t d = value[i + 1];
+        if (U16_IS_TRAIL(d)) {
```

Comment:
Why are there two flavors of macros? If I understand correctly, `U16_IS_SURROGATE_TRAIL` means "we already know it is a surrogate, now we are testing if it is the trail surrogate", and `U16_IS_TRAIL` means "we are testing if it is the trail surrogate". If so, the first one can just be replaced by the second one?